beacon-chain/core/helpers/validators.go

```diff
@@ -283,6 +259,49 @@ func BeaconProposerIndex(ctx context.Context, state state.ReadOnlyBeaconState) (
 	return ComputeProposerIndex(state, indices, seedWithSlotHash)
 }
 
+func BeaconProposerIndexFromCache(ctx context.Context, state state.ReadOnlyBeaconState, epoch primitives.Epoch) (primitives.ValidatorIndex, bool) {
+	e := time.CurrentEpoch(state)
+	// The cache uses the state root of the previous epoch - minimum_seed_lookahead last slot as key. (e.g. Starting epoch 1, slot 32, the key would be block root at slot 31)
+	// For simplicity, the node will skip caching of genesis epoch.
+	if e <= params.BeaconConfig().GenesisEpoch+params.BeaconConfig().MinSeedLookahead {
+		return 0, false
+	}
+
+	s, err := slots.EpochEnd(epoch)
+	if err != nil {
+		log.WithError(err).Error("could not get epoch end")
+		return 0, false
+	}
+	r, err := StateRootAtSlot(state, s)
+	if err != nil {
+		log.WithError(err).Error("could not get state root at slot")
+		return 0, false
+	}
+
+	if r == nil || bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
+		return 0, false
+	}
+
+	proposerIndices, err := proposerIndicesCache.ProposerIndices(bytesutil.ToBytes32(r))
+	if err != nil {
+		log.WithError(err).Error("could not interface with committee cache")
+		return 0, false
+	}
+	if proposerIndices == nil {
+		if err := UpdateProposerIndicesInCache(ctx, state); err != nil {
```

Comment:
If we are updating the cache, we should be able to retrieve the list of proposers